provider/cluster/monitor.go

```diff
@@ -148,26 +148,45 @@ func (m *deploymentMonitor) doCheck() (bool, error) {
 	badsvc := 0
 
 	for _, spec := range m.mgroup.Services {
-		found := false
-		for _, svc := range status.Services {
-			if svc.Name != spec.Name {
-				continue
-			}
-			found = true
-
-			if uint32(svc.Available) < spec.Count {
+		service, foundService := status.Services[spec.Name]
+		if foundService {
+			if uint32(service.Available) < spec.Count {
 				badsvc++
 				m.log.Debug("service available replicas below target",
 					"service", spec.Name,
-					"available", svc.Available,
+					"available", service.Available,
 					"target", spec.Count,
 				)
 			}
 		}
 
-		if !found {
+		expectedPortCount := 0
+		foundPortCount := 0
+		for _, exposed := range spec.Expose {
```

Comment:
can we somehow refactor block down below? to many nested levels.

Reply:
I agree it is pretty deep, but I think we will refactor this whenever we add support for more endpoint features like individual public IP, etc.